server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -678,6 +654,21 @@ public XContentBuilder snapshotsToXContent(
             if (version != null) {
                 builder.field(VERSION, version.toString());
             }
+
+            if (snapshotDetails.getStartTimeMillis() != -1) {
+                builder.field(START_TIME_MILLIS, snapshotDetails.getStartTimeMillis());
+            }
+            if (snapshotDetails.getEndTimeMillis() != -1) {
+                builder.field(END_TIME_MILLIS, snapshotDetails.getEndTimeMillis());
+            }
+            if (snapshotDetails.getPartialIndices() != null) {
+                builder.startArray(PARTIAL_INDICES);
```

Comment:
Ugh, I thought we could get away with this but now I see that we can't. In older versions we will leniently skip over individual fields, but not a whole array of things :( Ok this needs more work.

Reply:
Ah fun timing :) I just posted a suggestion for this one below/above.